brainscore/benchmarks/objectnet.py

```diff
@@ -0,0 +1,59 @@
+import os
+import logging
+
+import numpy as np
+import pandas as pd
+
+import brainscore
+from brainio.stimuli import StimulusSet
+from brainscore.benchmarks import BenchmarkBase
+from brainscore.metrics import Score
+from brainscore.metrics.accuracy import Accuracy
+from brainscore.model_interface import BrainModel
+from brainio.fetch import StimulusSetLoader
+from brainio.lookup import lookup_stimulus_set
+
+NUMBER_OF_TRIALS = 10
+
+_logger = logging.getLogger(__name__)
+LOCAL_STIMULUS_DIRECTORY = '/braintree/data2/active/common/objectnet-stimuli/'
+
+class Objectnet(BenchmarkBase):
+    def __init__(self):
+        self._stimulus_set = brainscore.get_stimulus_set('katz.BarbuMayo2019')
+        self._similarity_metric = Accuracy()
+        ceiling = Score([1, np.nan], coords={'aggregation': ['center', 'error']}, dims=['aggregation'])
+       
+        super(Objectnet, self).__init__(identifier='katz.BarbuMayo2019-top1', version=1,
+                                           ceiling_func=lambda: ceiling,
+                                           parent='ObjectNet',
+                                           bibtex="""@inproceedings{DBLP:conf/nips/BarbuMALWGTK19,
+                                                    author    = {Andrei Barbu and
+                                                                David Mayo and
+                                                                Julian Alverio and
+                                                                William Luo and
+                                                                Christopher Wang and
+                                                                Dan Gutfreund and
+                                                                Josh Tenenbaum and
+                                                                Boris Katz},
+                                                    title     = {ObjectNet: {A} large-scale bias-controlled dataset for pushing the
+                                                                limits of object recognition models},
+                                                    booktitle = {NeurIPS 2019},
+                                                    pages     = {9448--9458},
+                                                    year      = {2019},
+                                                    url       = {https://proceedings.neurips.cc/paper/2019/hash/97af07a14cacba681feacf3012730892-Abstract.html},
+                                                    }""")
+
+
+    def __call__(self, candidate):
+        # The proper `fitting_stimuli` to pass to the candidate would be the imagenet training set.
+        # For now, since almost all models in our hands were trained with imagenet, we'll just short-cut this
+        # by telling the candidate to use its pre-trained imagenet weights.
+        candidate.start_task(BrainModel.Task.label, 'imagenet')
```

Comment:
is there perfect overlap between imagenet labels and objectnet labels? If not, should this be taken into account in the similarity metric / or even better when instructing the model about the task?

Reply:
the synset labels for the objectnet categories we used are perfectly aligned with the ImageNet ones. is that what you're asking?